data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3Service.java

```diff
@@ -68,4 +75,20 @@ S3Client createS3Client() {
                         .build())
                 .build();
     }
+    //Issue#1971-Start
+    /**
+     * Create a S3AsyncClient Object for S3 Select query
+     * @return a S3AsyncClient Object
+     */
+    private S3AsyncClient createS3AsyncClient() {
```

Comment:
Thanks, David for your review. 
As per our observation, we are unable to find the s3 select feature calling methods in the S3Client class in SDK version 2x, which is used in data prepper.
whereas S3AsyncClient class contains the S3 select feature using selectObjectContent method.
In addition, we discovered the following information in the AWS documentation for the S3 Select example, which is built on SDK version 1. This is also not helpful because our data prepper is using SDK version 2X. So, I went through the SDK code on GitHub and discovered that #2943 provided the S3 Select Object feature in s3AsyncClient. I've attached the aws-sdk-v2 issue link (859) and the fix link (2943) for your reference.
Links:
https://github.com/aws/aws-sdk-java-v2/issues/859
https://github.com/aws/aws-sdk-java-v2/issues/2943
AWS SDK Version 1 Example Link:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-select.html
Please provide your feedback on this.

docs.aws.amazon.comdocs.aws.amazon.com
[Examples of using Amazon S3 Select on objects - Amazon Simple Storage Service] (https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-select.html)
You can use S3 Select with the Amazon S3 REST API and the AWS SDK to select content from objects.

Reply:
Thank you for the clarification Uday. Based on your response, we can keep the async client.